execution/executor/tests/storage_integration_test.rs

```diff
@@ -247,6 +247,520 @@ fn test_reconfiguration() {
+        .unwrap();
+    verify_committed_txn_status(t1.as_ref(), &block1[0]).unwrap();
+
+    // Transaction 2, 3 are rejected
+    let (t2, _) = response_items
+        .pop()
+        .unwrap()
+        .into_get_account_txn_by_seq_num_response()
+        .unwrap();
+    assert!(t2.is_none());
+
+    // Now that the PublishingOption is modified to CustomScript, we can resubmit the script again.
+
+    let txn2 = get_test_signed_transaction(
+        validator_account,
+        /* sequence_number = */ 0,
+        validator_privkey.clone(),
+        validator_pubkey.clone(),
+        Some(script1),
+    );
+
+    let txn3 = get_test_signed_transaction(
+        validator_account,
+        /* sequence_number = */ 1,
+        validator_privkey.clone(),
+        validator_pubkey.clone(),
+        Some(script2),
+    );
+
+    let block2 = vec![txn2, txn3];
+    let output2 = executor
+        .execute_block(
+            HashValue::zero(),
+            block2.clone(),
+            &committed_trees,
+            &committed_trees,
+        )
+        .unwrap();
+
+    let block2_id = gen_block_id(2);
+
+    let ledger_info_with_sigs = gen_ledger_info_with_sigs(4, output2.accu_root(), block2_id);
+    executor
+        .commit_blocks(
+            vec![(block2.clone(), Arc::new(output2))],
+            ledger_info_with_sigs,
+            &committed_trees,
+        )
+        .unwrap();
+
+    let request_items = vec![
+        RequestItem::GetAccountTransactionBySequenceNumber {
+            account: validator_account,
+            sequence_number: 0,
+            fetch_events: false,
+        },
+        RequestItem::GetAccountTransactionBySequenceNumber {
+            account: validator_account,
+            sequence_number: 1,
+            fetch_events: false,
+        },
+    ];
+
+    let (
+        mut response_items,
+        ledger_info_with_sigs,
+        validator_change_proof,
+        _ledger_consistency_proof,
+    ) = rt
+        .block_on(
+            storage_read_client.update_to_latest_ledger(
+                /* client_known_version = */ 0,
+                request_items.clone(),
+            ),
+        )
+        .unwrap();
+    let trusted_state = TrustedState::new_trust_any_genesis_WARNING_UNSAFE();
+    verify_update_to_latest_ledger_response(
+        &trusted_state,
+        0,
+        &request_items,
+        &response_items,
+        &ledger_info_with_sigs,
+        &validator_change_proof,
+    )
+    .unwrap();
+    response_items.reverse();
+
+    // Transaction 2 is committed.
+    let (t1, _) = response_items
+        .pop()
+        .unwrap()
+        .into_get_account_txn_by_seq_num_response()
+        .unwrap();
+    verify_committed_txn_status(t1.as_ref(), &block2[0]).unwrap();
+
+    // Transaction 3 is NOT committed.
+    let (t2, _) = response_items
+        .pop()
+        .unwrap()
+        .into_get_account_txn_by_seq_num_response()
+        .unwrap();
+    assert!(t2.is_none());
+}
+
```

Comment:
These tests are pretty long/hard to follow... would it be possible to abstract some of the common setup into a helper method and/or to add helper methods for the common pattern of execute/commit/check responses?

Reply:
It would be fine to do this in a follow-up too, but it seems like a very important problem to solve sometime soon. We don't want to add 500+ lines of tests for every new on-chain config...

Reply:
cc @msmouse. Do you have any suggestions?